drivers/stepper/adi_tmc/tmc51xx/tmc51xx_stepper_driver.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 Dipak Shetty
+ * SPDX-FileCopyrightText: Copyright (c) 2025 Jilay Sandeep Pandya
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/stepper.h>
+#include "tmc51xx.h"
+#include <adi_tmc5xxx_common.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(tmc51xx, CONFIG_STEPPER_LOG_LEVEL);
+
+#define DT_DRV_COMPAT adi_tmc51xx_stepper_driver
+
+struct tmc51xx_stepper_driver_config {
+	const uint16_t default_micro_step_res;
+	const int8_t sg_threshold;
+	/* parent controller required for bus communication */
+	const struct device *controller;
+};
+
+struct tmc51xx_stepper_data {
+	stepper_drv_event_cb_t drv_event_cb;
+	void *drv_event_cb_user_data;
+};
+
+void tmc51xx_driver_trigger_cb(const struct device *dev, const enum stepper_drv_event event)
+{
+	struct tmc51xx_stepper_data *data = dev->data;
+
+	if (!data->drv_event_cb) {
+		LOG_WRN_ONCE("No stepper driver callback registered");
+		return;
+	}
+	data->drv_event_cb(dev, event, data->drv_event_cb_user_data);
+}
+
+static int tmc51xx_set_stepper_drv_event_callback(const struct device *stepper,
+						  stepper_drv_event_cb_t callback, void *user_data)
+{
+	struct tmc51xx_stepper_data *data = stepper->data;
+
+	data->drv_event_cb = callback;
+	data->drv_event_cb_user_data = user_data;
+
+	return 0;
+}
+
+static int tmc51xx_stepper_enable(const struct device *dev)
```

Comment:
```suggestion
static int tmc51xx_stepper_driver_enable(const struct device *dev)
```
Also check other functions and macros in this file.